src/Functions/regexpExtract.cpp

```diff
@@ -124,21 +124,23 @@ class FunctionRegexpExtract : public IFunction
         res_offsets.push_back(res_offset);
     }
 
-    static void vectorConstant(
+    void vectorConstant(
         const ColumnString::Chars & data,
         const ColumnString::Offsets & offsets,
         const std::string & pattern,
         ssize_t index,
         ColumnString::Chars & res_data,
-        ColumnString::Offsets & res_offsets)
+        ColumnString::Offsets & res_offsets) const
     {
         const OptimizedRegularExpression regexp = Regexps::createRegexp<false, false, false>(pattern);
         unsigned capture = regexp.getNumberOfSubpatterns();
         if (index < 0 || index >= capture + 1)
             throw Exception(
                 ErrorCodes::INDEX_OF_POSITIONAL_ARGUMENT_IS_OUT_OF_RANGE,
-                "Index value {} is out of range, should be in [0, {})",
+                "Index value {} is out of range of regexp pattern '{}' in function {}, should be in [0, {})",
```

Comment:
The error message now sounds slightly weird. Maybe

```
Index value {} for regexp pattern `{}` in function {} is out-of-range, should be in [0, {})`?
```

Reply:
Sounds better. I'll fix it.